libs/ledger-live-common/src/families/cosmos/js-signOperation.ts

```diff
@@ -31,79 +30,47 @@ const signOperation = ({
       let cancelled;
 
       async function main() {
-        const hwApp = new Cosmos(transport);
-
-        const { accountNumber, sequence } = await defaultCosmosAPI.getAccount(
+        const cosmosAPI = new CosmosAPI(account.currency.id);
+        const { accountNumber, sequence } = await cosmosAPI.getAccount(
           account.freshAddress
         );
-
-        const chainId = await defaultCosmosAPI.getChainId();
-
         o.next({ type: "device-signature-requested" });
-
-        const { publicKey } = await hwApp.getAddress(
-          account.freshAddressPath,
-          "cosmos",
-          false
-        );
-
-        const pubkey = {
-          typeUrl: "/cosmos.crypto.secp256k1.PubKey",
-          value: new Uint8Array([
-            ...new Uint8Array([10, 33]),
-            ...new Uint8Array(Buffer.from(publicKey, "hex")),
-          ]),
-        };
-
-        const unsignedPayload = await buildTransaction(
-          account as CosmosAccount,
+        const { aminoMsgs, protoMsgs } = await buildTransaction(
+          account,
           transaction
         );
 
-        const msgs = unsignedPayload.map((msg) => aminoTypes.toAmino(msg));
-
+        const feeToEncode = {
+          amount: [
+            {
+              denom: account.currency.units[1].code,
+              amount: transaction.fees?.toString() as string,
```

Comment:
is it really as string ? what if fees is not found ? we will have undefined I think

Reply:
fixed